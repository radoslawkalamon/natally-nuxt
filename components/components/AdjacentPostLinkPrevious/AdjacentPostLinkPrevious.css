.adjacent-post-link-previous {
  color: inherit;
  display: flex;
  flex-flow: column nowrap;
  gap: var(--gap);
  max-width: var(--max-width-adjacent-post-link);
  text-align: left;
  text-decoration: none;
  width: 100%;
}

.adjacent-post-link-previous__label {
  font-size: var(--font-size-footnotes);
  line-height: var(--line-height-footnotes);
}

.adjacent-post-link-previous__label::before {
  color: var(--color-accent);
  content: var(--content-arrows-left) ' ';
}

.adjacent-post-link-previous__title {
  font-family: var(--font-family-sans);
  font-size: var(--font-size-header-3);
  font-weight: bold;
  line-height: var(--line-height-header);
  text-decoration: underline;
}
