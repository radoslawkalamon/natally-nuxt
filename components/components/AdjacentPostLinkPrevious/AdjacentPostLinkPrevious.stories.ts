import type { Meta, Story } from '@storybook/vue'
import ComponentsAdjacentPostLinkPrevious from '@/components/components/AdjacentPostLinkPrevious/AdjacentPostLinkPrevious.vue'

const meta: Meta = {
  title: 'Components / Adjacent Post Link Previous',
  component: ComponentsAdjacentPostLinkPrevious
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: {
    ComponentsAdjacentPostLinkPrevious
  },
  props: Object.keys(argTypes),
  template: '<ComponentsAdjacentPostLinkPrevious v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  label: 'Poprzedni',
  path: '/post/test-post/',
  title: 'Post testowy'
}
