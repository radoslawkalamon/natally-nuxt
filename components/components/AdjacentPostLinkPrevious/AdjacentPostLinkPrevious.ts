import Vue from 'vue'

export default Vue.extend({
  name: 'ComponentsAdjacentPostLinkPrevious',
  props: {
    label: {
      type: String,
      required: true
    },
    path: {
      type: String,
      required: true
    },
    title: {
      type: String,
      required: true
    }
  }
})
