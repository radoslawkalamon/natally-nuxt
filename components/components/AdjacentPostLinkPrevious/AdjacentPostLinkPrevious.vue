<template>
  <NuxtLink
    class="adjacent-post-link-previous"
    data-test="components-adjacent-post-link-previous"
    :to="path"
  >
    <span
      class="adjacent-post-link-previous__title"
      v-text="title"
    />
    <span
      class="adjacent-post-link-previous__label"
      v-text="label"
    />
  </NuxtLink>
</template>

<script src="./AdjacentPostLinkPrevious.ts" lang="ts"></script>
<style src="./AdjacentPostLinkPrevious.css" scoped></style>
