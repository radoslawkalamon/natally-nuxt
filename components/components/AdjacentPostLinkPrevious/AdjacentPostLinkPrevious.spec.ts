import AdjacentPostLinkPrevious from '@/components/components/AdjacentPostLinkPrevious/AdjacentPostLinkPrevious.vue'
import { shallPassUnitSanityTest } from '@/devtools/jest.common.spec'

describe('Components / Adjacent Post Link Previous', () => {
  describe('Unit', () => {
    shallPassUnitSanityTest({
      component: AdjacentPostLinkPrevious,
      options: {
        propsData: {
          label: 'NastÄ™pny',
          path: '/post/test-path/',
          title: 'Post testowy'
        }
      }
    })
  })
})
