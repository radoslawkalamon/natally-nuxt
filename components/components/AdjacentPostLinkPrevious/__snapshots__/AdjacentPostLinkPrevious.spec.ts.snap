// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Adjacent Post Link Previous Unit shall pass unit sanity test 1`] = `"<nuxtlink-stub to=\\"/post/test-path/\\" class=\\"adjacent-post-link-previous\\"><span class=\\"adjacent-post-link-previous__title\\">Post testowy</span> <span class=\\"adjacent-post-link-previous__label\\">NastÄ™pny</span></nuxtlink-stub>"`;
